plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +70,187 @@ def test_plot_empty_data(self):
         py.sign_in('PlotlyImageTest', '786r5mecv0')
         self.assertRaises(PlotlyEmptyDataError, py.plot, [],
                           filename='plot_invalid')
+
+    def test_plot_sharing_argument(self):
+
+        # Raise an error if sharing argument is incorrect
+        # correct arguments {'public, 'private', 'secret'}
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-sharing-argument',
+                  'sharing': 'privste'}
+
+        with self.assertRaisesRegexp(PlotlyError, 'sharing'):
+            py.plot(fig, **kwargs)
+
+    def test_plot_world_readable_sharing_conflict_1(self):
+
+        # Raise an error if world_readable=False but sharing='public'
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-privacy-setting',
+                  'world_readable': False,
+                  'sharing': 'public'}
+
+        with self.assertRaisesRegexp(PlotlyError, 'sharing'):
+            py.plot(fig, **kwargs)
+
+    def test_plot_world_readable_sharing_conflict_2(self):
+
+        # Raise an error if world_readable=True but sharing='secret'
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
```

Comment:
Looks like you use this *same* figure a lot, could you consider putting it in the `TestCase.setUp` method?

```
class TestPlot(TestCase):

    def setUp(self):
        self.simple_figure = {'data': [{'x': [1, 2, 3], 'y': [2, 1, 2]}]}
        super(TestPlot, self).setUp()

    # yadda, yadda
```

`unittest` will run `setUp` *before* each test method.

Making changes like these makes it easier to distinguish setup-logic from testing logic.